keyboards/misterknife/knife66/config.h

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2015 Jun Wako <wakojun@gmail.com>
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xAC11
+#define PRODUCT_ID      0x4172
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    MisterKnife
+#define PRODUCT         Knife66
+#define DESCRIPTION     Knife66 Keyboard
+
+#define MATRIX_ROWS 5
+#define MATRIX_COLS 15
+
+#define MATRIX_COL_PINS { B2, B1, B0, A7, A6, A5, A4, B9, B8, B7, B6, B5, B4, B3, A15 }
+#define MATRIX_ROW_PINS { B15, A8, A3, A2, A1 }
+#define DIODE_DIRECTION COL2ROW
+
+#define RGB_DI_PIN A9
```

Comment:
Should look into the PWM WS2812 driver as this can use TIM1_CH2 and give better end user XP

Reply:
will look into the WS2812 driver on the next board. I cannot change the pinout anymore unfortunately so it's no option

Reply:
You shouldn't need to change the pinout - from the F072 datasheet, section 5, table 15 (alternate functions for port A):
![image](https://user-images.githubusercontent.com/4781841/89024218-16c09200-d368-11ea-94e6-3a83d51be01e.png)

Reply:
Ah yes.

I've tried it with this: 
```
#define WS2812_PWM_DRIVER PWMD1  // default: PWMD2
#define WS2812_PWM_CHANNEL 2  // default: 2
#define WS2812_PWM_PAL_MODE 2  // Pin "alternate function", see the respective datasheet for the appropriate values for your MCU. default: 2
#define WS2812_DMA_STREAM STM32_DMA1_STREAM3  // DMA Stream for TIMx_UP, see the respective reference manual for the appropriate values for your MCU.
#define WS2812_DMA_CHANNEL 3  // DMA Channel for TIMx_UP, see the respective reference manual for the appropriate values for your MCU.
```
and activated it in the mcu and hal configs. But the LEDs don't seem to be going on, unfortunately.
I looked at my LEDs and turns out they do not support WS2812

[IN-PI554FCH](http://www.inolux-corp.com/datasheet/SMDLED/Addressable%20LED/IN-PI554FCH.pdf)

Reply:
@fauxpark could you assist, I'm kinda stuck

Reply:
@afewyards have you enabled PWM in the hardware config?  If not, then this won't work, outright.

Reply:
Yes I have.

Reply:
@zvecr can we move this along?

Reply:
Without the ws2812 enabled they are working. And in the datasheet it’s stated that they have an integrated ic.

I tried with tim1 enabled. But yeah no luck.

Reply:
@drashna can we merge?

Reply:
![image](https://user-images.githubusercontent.com/2985843/93135694-19abf200-f71e-11ea-88e0-438b6260ddcb.png)
`TIM1_UP` is channel 5 -- you've got it set to channel 3 above.
Sometimes ChibiOS starts from `0` despite the datasheet starting from `1`, so you may need to set either:
```c
#define WS2812_DMA_CHANNEL 5  // DMA Channel for TIMx_UP, see the respective reference manual for the appropriate values for your MCU.
```
or, if ChibiOS decides to start from `0`:
```c
#define WS2812_DMA_CHANNEL 4  // DMA Channel for TIMx_UP, see the respective reference manual for the appropriate values for your MCU.
```

Reply:
well I be damned. that was it! thanks @tzarc